rust/main/chains/hyperlane-sealevel/src/log_meta_composer.rs

```diff
@@ -64,15 +63,17 @@ impl LogMetaComposer {
             )))?
         }
 
-        let (transaction_index, transaction_hash) =
-            transaction_hashes
-                .into_iter()
-                .next()
-                .ok_or(HyperlaneSealevelError::NoTransactions(format!(
+        let (transaction_index, transaction_hash, program_index) = transaction_hashes
+            .into_iter()
+            .next()
+            .ok_or(HyperlaneSealevelError::NoTransactions(format!(
                 "block which should contain {} transaction does not contain any after filtering",
                 self.transaction_description,
             )))?;
 
+        // Construct log index which will be increasing relative to block
+        let log_index = U256::from((transaction_index << 8) + (program_index as usize));
```

Comment:
Noting that because the log index is used to determine "uniqueness" of an IGP payment, changing this may have some consequences. If we re-index a payment, we'll consider them as two separate payments because the log index is different:
- scraper: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/main/agents/scraper/src/db/payment.rs#L58
- other agents: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/main/hyperlane-base/src/db/rocks/hyperlane_db.rs#L169-L181

What I'm not sure about is if we'd expect to actually double-process any gas payments?
On the relayer/validator side I'm pretty sure this won't happen, but unsure about the scraper.